pilot/pkg/model/config.go

```diff
@@ -569,15 +591,22 @@ func ResolveShortnameToFQDN(host string, meta ConfigMeta) string {
 	return out
 }
 
+type authCache struct {
+	cache cache.ExpiringCache
+}
+
 // istioConfigStore provides a simple adapter for Istio configuration types
 // from the generic config registry
 type istioConfigStore struct {
 	ConfigStore
+
+	// AuthCache is a cache for auth(right now jwks_uri), it has some expiration logic on updating the URIs.
+	authCache authCache
 }
 
 // MakeIstioStore creates a wrapper around a store
 func MakeIstioStore(store ConfigStore) IstioConfigStore {
-	return &istioConfigStore{store}
+	return &istioConfigStore{store, authCache{cache.NewTTL(jwksURICacheExpiration, jwksURICacheEviction)}}
```

Comment:
please move authCache and related const etc to another file, such as auth_cache.go and write a auth_cache_test.go.
add a makeAuthCache function and use it here.

Reply:
done.